src/api/app.py

```diff
@@ -24,139 +24,146 @@
-def get_classes():
+@app.get('/api/class')
+@cache(expire=Constants.HOUR_IN_SECONDS, coder=PickleCoder, namespace="API_CACHE")
+async def get_classes(request: Request, semester: str or None = None, search: str or None = None):
     """
     GET /api/class?semester={}&search={}
     Cached: 1 Hour
     """
-    semester = request.args.get("semester", default=None)
-    search  = request.args.get("search", default=None)
     if semester:
         if not semester_info.is_public(semester):
-            if is_admin_user():
+            if is_admin_user(request.session):
                 classes, error = class_info.get_classes_full(semester)
-                return jsonify(classes) if not error else Response(error, status=500)
-            return Response("Semester isn't available", status=401)
+                return classes if not error else Response(error, status_code=500)
+            return Response(content="Semester isn't available", status_code=401)
         if search is not None:
             classes, error = class_info.get_classes_by_search(semester, search)
         else:
             classes, error = class_info.get_classes_full(semester)
-        return jsonify(classes) if not error else Response(error, status=500)
-    return Response("missing semester option", status=400)
-@app.route('/api/department', methods=['GET'])
-@cache.cached(timeout=Constants.HOUR_IN_SECONDS)
-def get_departments():
+        return classes if not error else Response(error, status_code=500)
+    return Response(content="missing semester option", status_code=400)
+    
+@app.get('/api/department')
+@cache(expire=Constants.HOUR_IN_SECONDS, coder=PickleCoder, namespace="API_CACHE")
+async def get_departments():
     """
     GET /api/department
     Cached: 1 Hour
-
+ 
     List of departments i.e. COGS, CIVL, CSCI, BIOL
     """
     departments, error = class_info.get_departments()
-    return jsonify(departments) if not error else Response(error, status=500)
+    return departments if not error else Response(content=error, status_code=500)
 
-@app.route('/api/subsemester', methods=['GET'])
-@cache.cached(timeout=Constants.HOUR_IN_SECONDS, query_string=True)
-def get_subsemesters():
+@app.get('/api/subsemester')
+@cache(expire=Constants.HOUR_IN_SECONDS, coder=PickleCoder, namespace="API_CACHE")
+async def get_subsemesters(subsemester: SubsemesterPydantic = Depends(SubsemesterPydantic)):
     """
     GET /api/subsemester?semester={}
     Cached: 1 Hour
 
     Get list of departments i.e. COGS, CIVL, CSCI, BIOL
     (Used in dropdown in "Course Search"
     """
-    semester = request.args.get("semester", default=None)
-    if semester:
-        subsemesters, error = class_info.get_subsemesters(semester)
-        return jsonify(subsemesters) if not error else Response(error, status=500)
+    if subsemester.semester:
+        subsemesters, error = class_info.get_subsemesters(subsemester.semester)
+        # for i in subsemesters:
+        #     print(i)
```

Comment:
```suggestion
        #     print(i)
```
Unused code should be removed.